src/bin/add/main.rs

```diff
@@ -143,6 +147,34 @@ fn handle_add(args: &Args) -> Result<()> {
     }
     let deps = &args.parse_dependencies()?;
 
+    if let Some(ref requested_features) = args.features {
+        let available_features = deps
+            .iter()
+            .map(|d| d.available_features.clone())
+            .flatten()
+            .collect::<Vec<String>>();
+        let flattened_features: Vec<String> = requested_features
+            .iter()
+            .map(|feat| {
+                feat.split(' ')
```

Comment:
We should centralize this splitting, whether that happens before this point or if we do the validation after calling `set_features` (I'm leaning towards that but haven't checked how feasible it is)

Reply:
is that something you'd like to see addressed in this PR or can that be a separate one?

Reply:
If its just a question of us doing this validation on a `Dependency` after `set_features` has been called, then it would fit in this one.

Reply:
didn't quite figure out how to go the `set_features` route, but moving the split to before the check wasn't too bad so I went with that approach.